packages/backend-common/config.d.ts

```diff
@@ -20,6 +20,16 @@ export interface Config {
   };
 
   backend: {
+    /** Backend configuration for when request authentication is enabled */
+    auth?: {
+      /**
+       * Secret shared by all backends for generating tokens
+       * Format is base64 24-bit key
```

Comment:
@Rugvip any thoughts on this? Looks to me like we can consider accepting a JSON key structure to allow consumers to construct any flavour of key they like, rather than forcing a specific format ([here](https://github.com/backstage/backstage/pull/7943/files#diff-64f85cefd740255c28effb941c7d8de043fdad6ed1a73f55b5e1db5e366d56e7R42)). Has this been discussed at all so far?

Reply:
We discussed the length of the key and figured 24 *bytes* is a sane recommendation, there's nothing prohibiting a shorter or longer key though, so I think we could at least update this description to reflect that. Too short a key and it'll be insecure of course, and if the key is unnecessarily long it'll just be truncated.

We only kept symmetric keys in scope for now and didn't really discuss alternatives much there except that we might want to add other methods in the future.

When it comes to the algorithm used we went straight for `HS256`. It's likely to remain a solid choice for quite some time, but we could add additional algorithms if needed.

I could totally see us future proofing this a little bit though, because I'm pretty sure we'll want to add support for additional verification keys in the future, possibly with different format and key IDs. I don't think we want to use JWKs directly though, the format is a bit awkward. If we wanna wrap it up a little bit more we could perhaps do something like this? 

```yaml
backend:
  auth:
    keys:
      - secret: ...
```

With that we can add a defaulted `type: ` field to the keys in the future, making them symmetric by default. We would be signing with the first key in the array, and any key would be accepted during validation. As an optimization we could add an optional `kid` to each key for more explicit lookups, and we can also allow for a mix of symmetric and asymmetric keys of different formats.

Reply:
All makes sense. Address the bits/bytes typo in 51608d2021, and then pushed 189f7f8a72 taking a run at allowing for multiple keys and future-proofing the config a bit.